src/main/java/org/altbeacon/beacon/service/BeaconService.java

```diff
@@ -200,7 +202,14 @@ public void onCreate() {
                 BeaconManager.DEFAULT_FOREGROUND_BETWEEN_SCAN_PERIOD, mBackgroundFlag, mCycledLeScanCallback, bluetoothCrashResolver);
 
         beaconManager = BeaconManager.getInstanceForApplication(getApplicationContext());
-        beaconParsers = beaconManager.getBeaconParsers();
+
+        if (beaconManager.getBeaconParsers() != null) {
+            beaconParsers.addAll(beaconManager.getBeaconParsers());
```

Comment:
Does making beaconParsers a `Set` instead of a `List` help here?  Because there is no custom hashCode defined for `BeaconParser` all this will do is make sure that two identical objects are not added to the list.  Is this a concern?